waspc/data/Generator/templates/server/src/auth/providers/oauth/state.ts

```diff
@@ -64,54 +68,64 @@ export function validateAndGetOAuthState<OT extends OAuthType>({
   provider: ProviderConfig,
   req: ExpressRequest
 }): OAuthStateWithCodeFor<OT> {
-  return {
-    ...validateAndGetCode(req),
-    ...validateAndGetState(provider, req),
-    ...(oAuthType === 'OAuth2WithPKCE' && validateAndGetCodeVerifier(provider, req)),
+  const state: OAuthStateWithCodeFor<OT> = {
+    ...getCode(req),
+    ...getState(req),
+    ...(oAuthType === 'OAuth2WithPKCE' && getCodeVerifier(provider, req)),
   };
-}
 
-function generateAndStoreState(
-  provider: ProviderConfig,
-  res: ExpressResponse
-): { state: string } {
-  const state = generateState();
-  setOAuthCookieValue(provider, res, 'state', state);
+  validateOAuthState(provider, req, state);
 
-  return { state };
+  return state;
 }
 
-function generateAndStoreCodeVerifier(
+function storeOAuthStateInCookies<OT extends OAuthType>(
```

Comment:
I would leave the implementation details out of the function's name -> `storeOAuthState`.

Makes it easier to change storage mechanisms too, the beauty of encapsulation :)